kasa/iot/modules/dimmer.py

```diff
@@ -0,0 +1,245 @@
+                type=Feature.Type.Switch,
+                category=Feature.Category.Config,
+            )
+        )
+
+        self._add_feature(
+            Feature(
+                device=self._device,
+                container=self,
+                id="dimmer_gentle_off_time",
+                name="Dimmer Gentle Off Time",
+                icon="mdi:clock-in",
+                attribute_getter="gentle_off_time",
+                attribute_setter="set_gentle_off_time",
+                type=Feature.Type.Switch,
+                category=Feature.Category.Config,
+            )
+        )
+
+        self._add_feature(
+            Feature(
+                device=self._device,
+                container=self,
+                id="dimmer_gentle_on_time",
+                name="Dimmer Gentle On Time",
+                icon="mdi:clock-out",
+                attribute_getter="gentle_on_time",
+                attribute_setter="set_gentle_on_time",
+                type=Feature.Type.Switch,
+                category=Feature.Category.Config,
+            )
+        )
+
+        self._add_feature(
+            Feature(
+                device=self._device,
+                container=self,
+                id="dimmer_ramp_rate",
+                name="Dimmer Ramp Rate",
+                icon="mdi:clock-fast",
+                attribute_getter="ramp_rate",
+                attribute_setter="set_ramp_rate",
+                type=Feature.Type.Switch,
+                category=Feature.Category.Config,
+            )
+        )
+
+    def query(self) -> dict:
+        """Request Dimming configuration."""
+        req = merge(
+            self.query_for_command("get_dimmer_parameters"),
+            self.query_for_command("get_default_behavior"),
+        )
+
+        return req
+
+    @property
+    def config(self) -> dict:
+        """Return current configuration."""
+        return self.data["get_dimmer_parameters"]
+
+    @property
+    def threshold_min(self) -> int | None:
+        """Return the minimum dimming level for this dimmer."""
+        if (min := self.config.get("minThreshold")) is not None:
+            return int(min)
+        return None
+
+    async def set_threshold_min(self, min: int) -> dict:
+        """
+        Set the minimum dimming level for this dimmer.
+
+        The value will depend on the luminaries connected to the dimmer.
+
+        :param min: The minimum dimming level, in the range 0-51.
+        """
+        if (min < 0) or (min > 51):
+            raise KasaException(
+                "Minimum dimming threshold is outside the supported range: 0-51."
+            )
+        return await self.call("calibrate_brightness", {"minThreshold": min})
+
+    @property
+    def fade_off_time(self) -> int | None:
+        """Return the fade off animation duration."""
+        if (fade_time := self.config.get("fadeOffTime")) is not None:
+            return int(fade_time)
+        return None
```

Comment:
```suggestion
        return self.config.get("fadeOffTime")
```
Looking at the fixtures, most, if not all of values seem to be integers? If so, these can be simplified like this.

Reply:
They are all integers from what I can tell. I just forgot that the data dict was a json object not a string map, I'll remove the extra guards, since that cuts down on the branches.